esphome/components/display/__init__.py

```diff
@@ -55,6 +56,7 @@ def validate_rotation(value):
 BASIC_DISPLAY_SCHEMA = cv.Schema(
     {
         cv.Optional(CONF_LAMBDA): cv.lambda_,
+        cv.Optional(CONF_AUTO_CLEAR_ENABLED): cv.boolean,
```

Comment:
Please set the defaul here:
```suggestion
        cv.Optional(CONF_AUTO_CLEAR_ENABLED, default=True): cv.boolean,
```

Reply:
Thank you for the suggestion, I have applied that to my commit.

Reply:
Actually, I now remembered why I cannot do that. That would require adding support for the flag to all displays. The way I formulated this that can be avoided. Reverting to the original code.

Reply:
What do you mean? Isn't the flag already added to all displays by adding the field to `BASIC_DISPLAY_SCHEMA`? 

(I saw your commit before you reverted it and approved based on that, sorry about that).

Reply:
But by including it here in `BASIC_DISPLAY_SCHEMA` a user could set the flag on a non-supported display type and then get compile errors. 
The VSCode extension uses the schema to show valid options.

If this is only for pixel based displays then I suggest making a new schema extending from BASIC with this key and then those supported displays should extend that one.

Reply:
I think the problem was to add it to BASIC_DISPLAY_SCHEMA rather than into FULL_DISPLAY_SCHEMA, this meant that it was also used for, e.g., LCD-based displays.

I have now followed the example of rotation, which is also supported just for displays which use DisplayBuffer (and therefore moved the auto-clear config flag to FULL_DISPLAY_SCHEMA).

The test if the key exists in the config is still necessary, because also displays only using the basic schema invoke this through register_display, and thus they key may not always exist. This is the same as for rotation.

Reply:
Yeah, this makes more sense to me. Just needs a docs update then.